contracts/pact_state_handlers_methods.go

```diff
@@ -0,0 +1,170 @@
+
+	return nil, nil
+
+}
+
+func createComponents(setup bool, state models.ProviderState) (models.ProviderStateResponse, error) {
+	if !setup {
+		err := os.Setenv("SETUP", "false")
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		return nil, nil
+	}
+	err := os.Setenv("SETUP", "true")
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+	components := parseComponents(state.Parameters)
+	for _, comp := range components {
+		ghComp := getGhComponentSpec(comp.name, comp.app.namespace, comp.app.appName, comp.repo)
+
+		hasAppLookupKey := types.NamespacedName{Name: comp.app.appName, Namespace: comp.app.namespace}
+		createdHasApp := &appstudiov1alpha1.Application{}
+
+		//create gh component
+		gomega.Expect(k8sClient.Create(ctx, ghComp)).Should(gomega.Succeed())
+		hasCompLookupKey := types.NamespacedName{Name: comp.name, Namespace: comp.app.namespace}
+		createdHasComp := &appstudiov1alpha1.Component{}
+
+		// wait until component is created
+		gomega.Eventually(func() bool {
+			err := k8sClient.Get(context.Background(), hasCompLookupKey, createdHasComp)
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			return len(createdHasComp.Status.Conditions) > 1
+		}, timeout, interval).Should(gomega.BeTrue())
+
+		// wait until component is ready
+		gomega.Eventually(func() bool {
+			err := k8sClient.Get(context.Background(), hasAppLookupKey, createdHasApp)
+			gomega.Expect(err).ToNot(gomega.HaveOccurred())
+			return len(createdHasApp.Status.Conditions) > 0 && strings.Contains(createdHasApp.Status.Devfile, comp.name)
+		}, timeout, interval).Should(gomega.BeTrue())
+	}
+	return nil, nil
+
+}
+
+func removeAllInstacesInNamespace(namespace string, myInstance client.Object) {
+	objectKind := strings.Split(reflect.TypeOf(myInstance).String(), ".")[1]
+	remainingCount := getObjectCountInNamespace(objectKind, namespace)
+	if remainingCount == 0 {
+		return
+	}
+	// remove resources in namespace
+	err := k8sClient.DeleteAllOf(context.Background(), myInstance, client.InNamespace(namespace))
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
+
+	// watch number of resources existing
+	gomega.Eventually(func() bool {
+		objectKind := strings.Split(reflect.TypeOf(myInstance).String(), ".")[1]
+		remainingCount := getObjectCountInNamespace(objectKind, namespace)
+		println("Removing", objectKind, "instances from", namespace, "namespace. Remaining: ", remainingCount)
+		return remainingCount == 0
+	}, timeout, interval).Should(gomega.BeTrue())
+}
+
+func getObjectCountInNamespace(objectKind string, namespace string) int {
+	unstructuredObject := &unstructured.Unstructured{}
+
+	unstructuredObject.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   appstudiov1alpha1.GroupVersion.Group,
+		Version: appstudiov1alpha1.GroupVersion.Version,
+		Kind:    objectKind,
+	})
+
+	err := k8sClient.List(context.Background(), unstructuredObject, &client.ListOptions{Namespace: namespace})
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
```

Comment:
You are not printing the error in gomega expect. In case of debug 0 information give this block